reference/forms/types/choice.rst

```diff
@@ -83,40 +87,98 @@ code, while the **key** is what will be shown to the user.
     and will be removed in 3.0. To read about the old API, read an older version of
     the docs.
 
+.. note::
+
+    Pre selected choices will depend on the **data** passed to the field and
+    the values of the ``choices`` option. However submitted choices will depend
+    on the **string** matching the **choice**. In the example above, the default
+    values are incrementing integers because ``null`` cannot be casted to string.
+    You should consider it as well when dealing with ``empty_data`` option::
+
+        $builder->add('isAttending', 'choice', array(
+            'choices'  => array(
+                'Maybe' => null,
+                'Yes' => true,
+                'No' => false,
+            ),
+            'choices_as_values' => true,
+            'data' => true, // pre selected choice
```

Comment:
@yceruto I understand your comment but this is not about a bad practice here, it's about make a clear difference I'm not sure https://github.com/symfony/symfony-docs/pull/6265 is enough though.